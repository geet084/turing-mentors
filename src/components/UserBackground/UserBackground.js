import React, { Component } from 'react';

export class UserBackground extends Component {
  state = {
    background: '',
  }

  submitForm = (e) => {
    e.preventDefault();
    this.props.updateUserInfo([this.state, 'userSchedule']);
  }

  goBack = (e) => {
    e.preventDefault();
    this.props.updateUserInfo([this.state, 'userBio']);
  }

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value })
  }

  render() {
    const { background } = this.state;
    const mentor = this.props.user === 'Mentor';

    return (
      <div>
        <form onSubmit={this.submitForm} autoComplete='off'>
          <span className="pages">{this.props.user} Background info:</span>
          <textarea
            className="bg-info"
            type="text"
            placeholder="Tell us a bit about yourself"
            name='background'
            rows="18"
            cols="30"
            value={background}
            onChange={this.handleChange} />
        </form>
        <button className="prev-btn" onClick={this.goBack}>Back</button>
        <button className="next-btn" onClick={this.submitForm}>Next</button>
        <span className="pages">3 of {mentor ? '6' : '4'}</span>
      </div>
    )
  }
}

export default UserBackground;