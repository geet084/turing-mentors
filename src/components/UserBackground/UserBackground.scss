.bg-info {
  display: block;
  resize: none;
  background-color: rgba(255, 255, 255, .9);
  border: 2px solid rgba(139, 139, 139, .3);
  color: #adadad;
  border-radius: 10px;
  white-space: nowrap;
  padding: 5px;
  margin: 10px auto;
  &::placeholder {
    color: #adadad;
  }
}