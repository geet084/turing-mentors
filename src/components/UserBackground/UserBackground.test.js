import React from 'react';
import { shallow } from 'enzyme';
import { UserBackground } from './UserBackground';

describe('UserBackground', () => {
  let wrapper;
  const mockProps = {
    updateUserInfo: jest.fn()
  };

  beforeEach(() => {
    wrapper = shallow(<UserBackground {...mockProps} />);

  });

  it('should match the correct snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should register a change when the input is typed in', () => {
    const mockEvent = { target: { value: 'new info', name: 'background' } };
    const expected = {
      background: 'new info'
    };

    wrapper.find('.bg-info').simulate('change', mockEvent);
    expect(wrapper.state()).toEqual(expected);
  });

  it('should go back a page when the back button is clicked', () => {
    const mockEvent = { preventDefault: () => { } }

    wrapper.find('.prev-btn').simulate('click', mockEvent);
    expect(mockProps.updateUserInfo).toHaveBeenCalled()
  });

  it('should go to the page when the next button is clicked', () => {
    const mockEvent = { preventDefault: () => { } }

    wrapper.find('.next-btn').simulate('click', mockEvent);
    expect(mockProps.updateUserInfo).toHaveBeenCalled()
  });
});