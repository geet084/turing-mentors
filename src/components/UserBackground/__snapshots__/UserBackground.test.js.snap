// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBackground should match the correct snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserBackground
    updateUserInfo={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          autoComplete="off"
          onSubmit={[Function]}
        >
          <span
            className="pages"
          >
             Background info:
          </span>
          <textarea
            className="bg-info"
            cols="40"
            name="background"
            onChange={[Function]}
            placeholder="Tell us a bit about yourself"
            rows="18"
            type="text"
            value=""
          />
        </form>,
        <button
          className="prev-btn"
          onClick={[Function]}
        >
          Back
        </button>,
        <button
          className="next-btn"
          onClick={[Function]}
        >
          Next
        </button>,
        <span
          className="pages"
        >
          3 of 
          4
        </span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoComplete": "off",
          "children": Array [
            <span
              className="pages"
            >
               Background info:
            </span>,
            <textarea
              className="bg-info"
              cols="40"
              name="background"
              onChange={[Function]}
              placeholder="Tell us a bit about yourself"
              rows="18"
              type="text"
              value=""
            />,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                " Background info:",
              ],
              "className": "pages",
            },
            "ref": null,
            "rendered": Array [
              undefined,
              " Background info:",
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "bg-info",
              "cols": "40",
              "name": "background",
              "onChange": [Function],
              "placeholder": "Tell us a bit about yourself",
              "rows": "18",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Back",
          "className": "prev-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Back",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Next",
          "className": "next-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Next",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "3 of ",
            "4",
          ],
          "className": "pages",
        },
        "ref": null,
        "rendered": Array [
          "3 of ",
          "4",
        ],
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            autoComplete="off"
            onSubmit={[Function]}
          >
            <span
              className="pages"
            >
               Background info:
            </span>
            <textarea
              className="bg-info"
              cols="40"
              name="background"
              onChange={[Function]}
              placeholder="Tell us a bit about yourself"
              rows="18"
              type="text"
              value=""
            />
          </form>,
          <button
            className="prev-btn"
            onClick={[Function]}
          >
            Back
          </button>,
          <button
            className="next-btn"
            onClick={[Function]}
          >
            Next
          </button>,
          <span
            className="pages"
          >
            3 of 
            4
          </span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoComplete": "off",
            "children": Array [
              <span
                className="pages"
              >
                 Background info:
              </span>,
              <textarea
                className="bg-info"
                cols="40"
                name="background"
                onChange={[Function]}
                placeholder="Tell us a bit about yourself"
                rows="18"
                type="text"
                value=""
              />,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  " Background info:",
                ],
                "className": "pages",
              },
              "ref": null,
              "rendered": Array [
                undefined,
                " Background info:",
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "bg-info",
                "cols": "40",
                "name": "background",
                "onChange": [Function],
                "placeholder": "Tell us a bit about yourself",
                "rows": "18",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Back",
            "className": "prev-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Back",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Next",
            "className": "next-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Next",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "3 of ",
              "4",
            ],
            "className": "pages",
          },
          "ref": null,
          "rendered": Array [
            "3 of ",
            "4",
          ],
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
